src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -380,7 +393,17 @@ public void reconcileSubscriptionsWithSubscriptionService(String orgId, boolean
     CustomBatchIterator.batchStreamOf(stream, batchSize)
         .forEach(
             batch -> {
-              batch.forEach(dto -> syncSubscription(dto, Optional.empty()));
+              batch.forEach(
+                  dto -> {
+                    // Contract provided subscriptions will have a different start_date than what is
+                    // stored in Subscription SearchAPI
+                    var existingSubscription =
+                        subscriptionRepository
+                            .findBySubscriptionNumber(dto.getSubscriptionNumber())
+                            .stream()
+                            .findFirst();
```

Comment:
Can we replace this with Guava's [onlyElement](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/collect/MoreCollectors.html#onlyElement())? Using findFirst is what bit us on [SWATCH-2494](https://issues.redhat.com/browse/SWATCH-2494). See https://github.com/RedHatInsights/rhsm-subscriptions/pull/3302

Or if more than one record is expected/allowed, can we add an `order by` clause so that we get deterministic results?

Reply:
I think we could get more than one here in an expired/updated case. The query already has an order by so I think we are good:
 https://github.com/RedHatInsights/rhsm-subscriptions/blob/78378a3f319836695a5ce6395615b24d51c4eaa5/swatch-core/src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java#L74